2.6/root/usr/share/container-scripts/mongodb/replset-functions.sh

```diff
@@ -0,0 +1,97 @@
+# This file contains functions	 for replSet manipulation
+
+# cache_container_addr waits till the container gets the external IP address and
+# cache it to disk
+function cache_container_addr() {
+  echo -n "=> Waiting for container IP address ..."
+  local i
+  for i in $(seq ${MAX_ATTEMPTS}); do
+    if ip -oneline -4 addr show up scope global | grep -Eo '[0-9]{,3}(\.[0-9]{,3}){3}' > "${HOME}"/.address; then
+      echo " $(mongo_addr)"
+      return 0
+    fi
+    sleep ${SLEEP_TIME}
+  done
+  echo "Failed to get Docker container IP address." && exit 1
+}
+
+# container_addr returns the current container external IP address
+function container_addr() {
+  echo -n $(cat ${HOME}/.address)
+}
+
+# mongo_addr returns the IP:PORT of the currently running MongoDB instance
+function mongo_addr() {
+  echo -n "$(container_addr):${port}"
+}
+
+# endpoints returns list of IP addresses with other instances of MongoDB
+# To get list of endpoints, you need to have headless Service named 'mongodb'.
+# NOTE: This won't work with standalone Docker container.
+function endpoints() {
+  service_name=${MONGODB_SERVICE_NAME:-mongodb}
+  dig ${service_name} A +search +short 2>/dev/null
+}
+
+# replset_addr return the address of the current replSet
+function replset_addr() {
+  echo "${MONGODB_REPLICA_NAME}/$(echo $(endpoints) | sed -e 's| |:27017,|g' -e 's|$|:27017|')"
```

Comment:
echo removes the new lines from dig output and sed adds port number to each address and joins them with comma -> `repl_name/host1,host2,...`